source/expressions.tex

```diff
@@ -1443,21 +1443,34 @@
 \pnum
 The result is the entity denoted by
 the \grammarterm{unqualified-id}\iref{basic.lookup.unqual}.
-If the entity is a local entity\iref{basic.pre}
-and naming it from outside of an unevaluated operand
-within the scope where the \grammarterm{unqualified-id} appears
-would result in some intervening \grammarterm{lambda-expression}
-capturing it by copy\iref{expr.prim.lambda.capture},
+If the \grammarterm{unqualified-id} appears
+in a \grammarterm{lambda-expression} at program point $P$ and
+the entity is a local entity\iref{basic.pre} or a variable declared by
+an \grammarterm{init-capture}\iref{expr.prim.lambda.capture},
+then let $S$ be the \grammarterm{compound-expression} of
+the innermost enclosing \grammarterm{lambda-expression} of $P$.
+If naming the entity from outside of an unevaluated operand within $S$
+would refer to an entity
+captured by copy in some intervening \grammarterm{lambda-expression},
+then let $E$ be the innermost such \grammarterm{lambda-expression}, and:
+\begin{itemize}
+\item
+If $P$ is in $E$'s function parameter scope
+but not its \grammarterm{parameter-declaration-clause}, then
 the type of the expression is
 the type of a class member access expression\iref{expr.ref}
 naming the non-static data member
 that would be declared for such a capture
-in the object parameter\iref{dcl.fct} of the function call operator of
-the innermost such intervening \grammarterm{lambda-expression}.
+in the object parameter\iref{dcl.fct} of the function call operator of $E$.
```

Comment:
@BRevzin: The base of the diff has changed here following the "Deducing `this`" paper; could you kindly check if the new edit is still appropriate?

Reply:
Yeah this looks right. The point was just to name the innermost such intervening lambda `E` 👍

Reply:
Thanks a lot!